eclipse-theia/theia
# [Plugin-API] Apply ResourceLabelFormatter API

Signed-off-by: Igor Vinokur <ivinokur@redhat.com>



#### What it does

Apply [`registerResourceLabelFormatter`](https://github.com/microsoft/vscode/blob/3b55119a98f460dae77830f802fa25ea44deff2e/src/vs/vscode.proposed.d.ts#L380) vscode Plugin-API function. It allows plugins to register a custom resource label formatters that specify how to display URIs. The test plugin registers a formatter:
```
theia.workspace.registerResourceLabelFormatter({
            scheme: 'file',
            formatting: {
                label: '${path} (formated)',
                separator: '/',
                tildify: true
            }
        });
    }));
```
CQ: https://dev.eclipse.org/ipzilla/show_bug.cgi?id=22392
#### How to test

1. Download the [test plugin](https://github.com/vinokurig/resource-label-formatter-plugin/releases/download/0.0.1/resource-label-formatter-plugin-0.0.1.vsix) and put it to the `plugins` directory.
2. Start Theia workspace with some git project
3. Open some file.
4. Point the mouse cursor to the opened editor tab, see the tooltip:
![Screenshot from 2020-07-29 18-17-41](https://user-images.githubusercontent.com/7668752/88819136-4173e480-d1c8-11ea-9691-1daa90ad09c6.jpg)

5. Run `F1 => Register Resource label formatter`
6. Point the mouse cursor to the opened editor tab, see the tooltip that shows the formatted URI, which is added by the registered formatter:
![Screenshot from 2020-07-29 18-17-02](https://user-images.githubusercontent.com/7668752/88819352-90ba1500-d1c8-11ea-84f2-d4e6a13b4cf3.jpg)

7. Run `F1 => Unregister formatter` and see the original tooltip.
#### Review checklist

#### Reminder for reviewers

- as a reviewer, I agree to behave in accordance with [the review guidelines](https://github.com/theia-ide/theia/blob/master/doc/pull-requests.md#reviewing)

fixes https://github.com/eclipse-theia/theia/issues/7946
